OMCompiler/SimulationRuntime/c/simulation/solver/synchronous.c

```diff
@@ -60,7 +60,7 @@ void initSynchronous(DATA* data, threadData_t *threadData, modelica_real startTi
   for(i=0; i<data->modelData->nBaseClocks; i++) {
     for(j=0; j<data->simulationInfo->baseClocks[i].nSubClocks; j++) {
       assertStreamPrint(threadData, data->simulationInfo->baseClocks[i].subClocks[j].solverMethod != NULL, "Continuous clocked systems aren't supported yet.");
-      assertStreamPrint(threadData, rat2Real(data->simulationInfo->baseClocks[i].subClocks[j].shift) >= 0, "Shift of sub-clock is negative. Sub-clocks aren't allowed to fire before base-clock.");
+      assertStreamPrint(threadData, data->simulationInfo->baseClocks[i].subClocks[j].shift.num >= 0, "Shift of sub-clock is negative. Sub-clocks aren't allowed to fire before base-clock.");
```

Comment:
We ensure that a rational `num/den` has `den>0` in `makeRational`. So this should be enough.
We could also make `den` unsigned?

Reply:
> We could also make den unsigned?

Yeah I thought about that, but then `num` and `den` have different type but they get mixed up every time we calculate something so that would lead to many conversions I think. I could write a dedicated `compareRatInt` or something.